src/Core/Environment.php

```diff
@@ -151,4 +166,93 @@ public static function getTimeLimitMax()
     {
         return static::$timeLimitMax;
     }
+
+    /**
+     * Get value of environment variable
+     *
+     * @param string $name
+     * @return mixed Value of the environment variable, or null if not set
+     */
+    public static function getEnv($name)
+    {
+        switch (true) {
+            case array_key_exists($name, static::$env):
+                return static::$env[$name];
+            case array_key_exists($name, $_ENV):
+                return $_ENV[$name];
+            case array_key_exists($name, $_SERVER):
+                return $_SERVER[$name];
+            default:
+                return getenv($name);
+        }
+    }
+
+    /**
+     * Set environment variable using php.ini syntax.
+     * Acts as a process-isolated version of putenv()
+     * Note: This will be parsed via parse_ini_string() which handles quoted values
+     *
+     * @param string $string Setting to assign in KEY=VALUE or KEY="VALUE" syntax
+     */
+    public static function putEnv($string)
+    {
+        // Parse name-value pairs
+        $envVars = parse_ini_string($string) ?: [];
+        foreach ($envVars as $name => $value) {
+            self::setEnv($name, $value);
+        }
+    }
+
+    /**
+     * Load environment variables from .env file
+     *
+     * @param string $path Path to the file
+     * @param bool $overload Set to true to allow vars to overload. Recommended to leave false.
+     * @return array List of values parsed as an associative array
+     */
+    public static function putEnvFromFile($path, $overload = false)
+    {
+        // Parse and cleanup content
+        $contents = implode(
+            PHP_EOL,
+            array_filter(array_map(
+                function ($line) {
+                    // Strip all # comments, including start or end of string, but not in quoted strings
+                    // '#comment', 'var=thing#comment', or 'var="not#comment"#comment
+                    return preg_replace('/((^[^"]*)|(^[^"]+("[^"]*){2}))#.*$/', '$1', $line);
+                },
+                explode(PHP_EOL, file_get_contents($path))
+            ))
+        );
+
+        // Pass to parse_ini_string
+        $variables = parse_ini_string($contents) ?: [];
```

Comment:
Actually, this is not ideal, as `#` is not a comment marker; `;` is, as per ini style.
As we discussed elsewhere, I'd leave inline # behaviour undefined, and only modify beginning of line comments as per https://github.com/NightJar/sapphire/blob/e614edec61f396c7e8f29bd5ee3340f56cf0b1a6/src/Core/Environment.php#L35-L36

This would reduce complexity, maintenance, and 'regex soup' that others have also raised.